### ISSUE TITLE
Factory reports

### ISSUE THREAD

**Issue by parafoxia** [parafoxia/analytix#9]
**Summary**
Factory methods can be used to simplify the process of creating more common reports. For example, the current method of getting the daily performance for a specific video is:

```py
client = YouTubeAnalytics("secrets.json")
client.retrieve(
    dt.date(2021, 1, 1),
    dt.date(2021, 6, 30),
    dimensions=("day",),
    filters={"video": "videoID"},
)
```
The proposed method would look (something) like this:
```py
factory = YouTubeAnalyticsFactory("secrets.json")
factory.video_performance("videoID", by="day", since=dt.date(2021, 1, 1))
```
A better example would be getting the daily performance channel wide:
```py
factory = YouTubeAnalyticsFactory("secrets.json")
factory.daily_performance(since=dt.date(2021, 1, 1))
```

**The idea**
It will make it quicker and easier for users to make more common requests.

**The ideal implementation**
A separate `YouTubeAnalyticsFactory` class.

**Contribution**
Do you want to contribute this feature yourself?
Yes

**Comment by parafoxia** [comment]
I'm changing the proposed implementation.

- The factory methods will be part of the `YouTubeAnalytics` class -- creating a new class for them seems a bit silly.
- There'll be three methods:

### Daily stats
```py
client.daily_stats(of=None, since=None, last=28, metrics="all")
```
where `of` is a video ID, `since` is the date to start collecting data from, `last` is the number of days to get data from (ignored if since is provided), and `metrics` is a series of metrics.

### Monthly stats
```py
client.monthly_stats(of=None, since=None, last=3, metrics="all")
```
where `of` is a video ID, `since` is the date to start collecting data from, `last` is the number of months to get data from (ignored if since is provided), and `metrics` is a series of metrics.

### Regional stats
```py
client.regional_stats(of=None, for_provinces=False, metrics="all")
```
where `of` is a video ID, `for_provinces` is whether data should be collected for provinces (states), and `metrics` is a series of metrics.

### Top videos
```py
client.top_videos(by="views", since=None, last=28, metrics="all")
```
where `by` is how to sort videos, `since` is the date to start collecting data from, `last` is the number of months to get data from (ignored if since is provided), and `metrics` is a series of metrics.

### Tasks
- [ ] `daily_stats`
- [ ] `monthly_stats`
- [ ] `regional stats`
- [ ] `top_videos`

**Comment by parafoxia** [comment]
Closed by 49928a4.
